logstash-core/src/main/java/org/logstash/common/LsQueueUtils.java

```diff
@@ -0,0 +1,83 @@
+package org.logstash.common;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+import org.logstash.ext.JrubyEventExtLibrary;
+
+/**
+ * Utilities around {@link BlockingQueue}.
+ */
+public final class LsQueueUtils {
+
+    private LsQueueUtils() {
+        //Utility Class
+    }
+
+    /**
+     * <p>Drains {@link JrubyEventExtLibrary.RubyEvent} from {@link BlockingQueue} with a timeout.</p>
+     * <p>The timeout will be reset as soon as a single {@link JrubyEventExtLibrary.RubyEvent} was
+     * drained from the {@link BlockingQueue}. Draining {@link JrubyEventExtLibrary.RubyEvent}
+     * stops as soon as either the required number of {@link JrubyEventExtLibrary.RubyEvent}s
+     * were pulled from the queue or the timeout value has gone by without an event drained.</p>
+     * @param queue Blocking Queue to drain {@link JrubyEventExtLibrary.RubyEvent}s
+     * from
+     * @param count Number of {@link JrubyEventExtLibrary.RubyEvent}s to drain from
+     * {@link BlockingQueue}
+     * @param nanos Timeout in Nanoseconds
+     * @return Collection of {@link JrubyEventExtLibrary.RubyEvent} drained from
+     * {@link BlockingQueue}
+     * @throws InterruptedException On Interrupt during {@link BlockingQueue#poll()} or
+     * {@link BlockingQueue#drainTo(Collection)}
+     */
+    public static Collection<JrubyEventExtLibrary.RubyEvent> drain(
+        final BlockingQueue<JrubyEventExtLibrary.RubyEvent> queue, final int count, final long nanos
```

Comment:
timeout would probably be a better name than nanos?

Reply:
@ph my reasoning here was that using `ns` as a timeout explicitly is a pretty nonstandard thing in Java.
So I thought I'd do what JDK does in a few spots and simply name it explicitly.

e.g. this in `ArrayBlockingQueue`

```java
    public boolean offer(E e, long timeout, TimeUnit unit)
        throws InterruptedException {

        checkNotNull(e);
        long nanos = unit.toNanos(timeout);
```

Saves you from having to read the JavaDoc imo :)